controller/src/main/java/io/pravega/controller/server/security/auth/handler/impl/AccessControlEntry.java

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.controller.server.security.auth.handler.impl;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Strings;
+import io.pravega.auth.AuthHandler;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.NonNull;
+
+import java.util.regex.Pattern;
+
+/**
+ * An entry of an {@link AccessControlList}.
+ */
+class AccessControlEntry {
+    private static final Pattern PATTERN_STAR_NOT_PRECEDED_BY_DOT = Pattern.compile("(?!=.)\\*");
+
+    @Getter(AccessLevel.PACKAGE)
+    private final String resourcePattern;
+
+    @Getter(AccessLevel.PACKAGE)
+    private final AuthHandler.Permissions permissions;
+
+    @VisibleForTesting
+    AccessControlEntry(@NonNull String aceResource, @NonNull AuthHandler.Permissions permissions) {
+        this(aceResource, permissions, false);
+    }
+
+    @VisibleForTesting
+    AccessControlEntry(@NonNull String aceResource, @NonNull AuthHandler.Permissions permissions, boolean isLegacyFormat) {
+        // Replaces any `*` with `.*`, if it's not already preceded by `.`, for regex processing.
+        // So, `pravega:://*` becomes `pravega:://.*` and `pravega:://scope:*` becomes `pravega:://scope:.*`
+        this.resourcePattern = isLegacyFormat ?  aceResource :
+                PATTERN_STAR_NOT_PRECEDED_BY_DOT.matcher(aceResource).replaceAll(".*");
+        this.permissions = permissions;
+    }
+
+    static AccessControlEntry fromString(String ace) {
+        String[] splits = null;
+        if (Strings.isNullOrEmpty(ace) || (splits = ace.split(",")).length != 2) {
```

Comment:
in which situations can "ace" split not be of length 2? and is that an illegal argument or is ace = null legit if the number of tokens in it are 3 or 1?

Reply:
Since ACEs are specified manually via the Password Auth Handler input file, they could be ill-formed. 

A couple of examples of the split being of a length != 2 are as follows: 
* `prn::/` (No comma - permission missing, splits = 1)
* `,,READ`  (3 splits)

Reply:
A minor point, but shouldn't it be an exception rather returning null? We expect a valid string, and if the string is not valid, then I'd say it is an exception.

Reply:
@fpj Returning `null` here makes the calling code in the `PasswordAuthhandler` much simpler, which is why I took this approach here. This method returns a non-null `AccessControlEntry` object for valid strings and null for invalid strings.